### PR TITLE
Story/group information management

### PR DESCRIPTION
**Pull request by Julian-Caleb**
## Story/Task

- [FE] Group Management 
https://trello.com/c/ZIeFvhIg
- [FE] Group Information Page
https://trello.com/c/WzHPdEGV

## Summary

- Add Group Management Page for mentor
- Add Group Information Page for admin

## Screenshot

- Group Information Page (null)
Screenshot (1530)

- Group Information Page for Desktop
Screenshot (1532)

- Group Management Page for Desktop
Screenshot (1533)

### CODE REVIEW

**Comment by msyahrulsp** [comment]
Branchnya tolong dipindah ke `staging` dlu @Julian-Caleb

**Comment by Julian-Caleb** [comment]
> Branchnya tolong dipindah ke `staging` dlu @Julian-Caleb

Maksudnya bagaimana Kak?

**Comment by msyahrulsp** [comment]
> Maksudnya bagaimana Kak?

Yang sekarang target branch buat mergenya itu `master`, harusnya ke `staging` wkwkwkwk

**Comment by Julian-Caleb** [comment]
Oh oke Kak

**Review by msyahrulsp — CHANGES_REQUESTED** [review]
- Buat semua placeholder bisa dihapus aja, biar nanti tinggal pasang layout yang udah ada
- Perhatiin juga ukurant font dipagenya, sesuain stylenya juga sama yang ada di figma
- Yang tooltip detail tugas aku cek belum ada juga, jadi jangan lupa bagian itu

**Comment by Julian-Caleb** [comment]
## Story/Task

- [FE] Group Management 
https://trello.com/c/ZIeFvhIg
- [FE] Group Information Page
https://trello.com/c/WzHPdEGV

## Summary

- Fix / refactor Group Management Page for mentor
- Fix / refactor Information Page for admin
- Add description for task and attendance

## Screenshot

- Group Information Page (null)
Screenshot (1562)
Screenshot (1564)

- Group Information Page for Desktop
Screenshot (1565)
Screenshot (1568)

- Group Management Page for Desktop
Screenshot (1569)
Screenshot (1571)

**Comment by Julian-Caleb** [comment]
## Story/Task

- [FE] Group Management 
https://trello.com/c/ZIeFvhIg
- [FE] Group Information Page
https://trello.com/c/WzHPdEGV

## Summary

- Fix / refactor and add Group Management Page for mentor to pages
- Fix / refactor and add Group Information Page for admin to pages
- Add description for task and attendance

## Screenshot

- Group Information Page (null)
Screenshot (1573)

- Group Information Page for Desktop
Screenshot (1575)
Screenshot (1576)

- Group Management Page for Desktop
Screenshot (1577)
Screenshot (1578)

File: src/pages/group-management.tsx [hunk 1]
```diff
@@ -1,7 +1,99 @@
...
+            mr={{ base: 0, md: 4 }}
+            w={{ base: '100%', md: 'auto' }}
+          >
+            <Image
+              src='/images/placeholder_logo_manage.png'
+              objectFit='contain'
+              height='100%'
+              alt='placeholder_logo_manage'
+            />
+          </Box>
```
Comment (msyahrulsp) [inline src/pages/group-management.tsx:46]: Pake komponen header dari staging

File: src/pages/group-management.tsx [hunk 2]
```diff
@@ -1,7 +1,99 @@
...
+
+          {/* Kelompok */}
+          <Box
+            width={{ base: '370px', md: '400px' }}
+            height='96px'
+            bgImage='/images/comet_container.png'
+            backgroundSize='cover'
+            borderRadius='25px'
+          >
```
Comment (msyahrulsp) [inline src/pages/group-management.tsx:55]: widthnya ga perlu di set manual. terus kasih bikin supaya centered vertically sama komponen header nanti

File: src/pages/group-management.tsx [hunk 3]
```diff
@@ -1,7 +1,99 @@
...
+              mr={12}
+              justifyContent='left'
+              alignItems='center'
+              height='100%'
+            >
+              <Image
+                src='/images/placeholder_profile.png'
+                alt='placeholder_profile'
+                mr={10}
+              />
```
Comment (msyahrulsp) [inline src/pages/group-management.tsx:67]: Pake [Avatar](https://chakra-ui.com/docs/components/avatar/usage)

**Review by msyahrulsp — CHANGES_REQUESTED** [review]
Ini tolong diubah ya. Konekin juga sekalian sama be sama hapus sekalian gambar2 placeholdernyaa 🫡🫡

File: src/pages/group-information.tsx [hunk 4]
```diff
@@ -1,7 +1,148 @@
...
+            mr={{ base: 0, md: 4 }}
+            w={{ base: '100%', md: 'auto' }}
+          >
+            <Image
+              src='/images/placeholder_logo_info.png'
+              objectFit='contain'
+              height='100%'
+              alt='placeholder_logo_info'
+            />
+          </Box>
```
[the remark this answers is not in the document]
Comment (Julian-Caleb) [reply src/pages/group-information.tsx:64]:
```suggestion
          <Header title='Group Information' />
```

**Comment by Julian-Caleb** [comment]
Saya usahakan ya Kak, mohon bantuannya :)

File: src/pages/group-management.tsx (hunk 1 above)
Comment (Julian-Caleb) [reply src/pages/group-management.tsx:46]:
```suggestion
          <Header title='Group Management' />
```

File: src/component/group-information-management/MentorBox.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  picture: string;
+  name: string;
+  nim: string;
+  faculty: string;
+}
+
+const MentorBox: React.FC<{ mentor: Mentor }> = ({ mentor }) => {
+  return (
+    <Flex
+      maxH='96px'
```
[the remark this answers is not in the document]
Comment (Julian-Caleb) [reply src/component/group-information-management/MentorBox.tsx:14]:
```suggestion
      <Flex
        width='480px'
        minW='200px'
        bg='#0B0A0A'
        p={4}
        borderRadius='25px'
        boxShadow='0 3px 15px -3px #FFFC83'
        color='white'
       alignItems='center'
    >
```

File: src/pages/group-information.tsx [hunk 6]
```diff
@@ -1,7 +1,148 @@
...
+                />
+                <Text fontWeight='bold' fontSize='24px'>
+                  {selectedOption}
+                </Text>
+                <MenuButton
+                  onClick={(e) => {
+                    e.stopPropagation();
+                  }}
+                >
+                  <Image src='/images/dropdown_icon.png' alt='dropdown_icon' />
```
[the remark this answers is not in the document]
Comment (Julian-Caleb) [reply src/pages/group-information.tsx:97]:
```suggestion
                  <RiArrowDownSLine size={24} />
```

File: src/component/group-information-management/MembersTable.tsx [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+          ) : (
+            <svg
+              xmlns='http://www.w3.org/2000/svg'
+              viewBox='0 0 128 128'
+              fill='#9B9B9B'
+              width='24px'
+              height='24px'
+            >
+              <path d='M 64 6 C 32 6 6 32 6 64 C 6 96 32 122 64 122 C 96 122 122 96 122 64 C 122 32 96 6 64 6 z M 64 12 C 92.7 12 116 35.3 116 64 C 116 92.7 92.7 116 64 116 C 35.3 116 12 92.7 12 64 C 12 35.3 35.3 12 64 12 z M 85.037109 48.949219 C 84.274609 48.974219 83.500391 49.300391 82.900391 49.900391 L 62 71.599609 L 51.099609 59.900391 C 49.999609 58.700391 48.100391 58.599219 46.900391 59.699219 C 45.700391 60.799219 45.599219 62.700391 46.699219 63.900391 L 59.800781 78 C 60.400781 78.6 61.1 79 62 79 C 62.8 79 63.599219 78.699609 64.199219 78.099609 L 87.199219 54 C 88.299219 52.8 88.299609 50.900781 87.099609 49.800781 C 86.549609 49.200781 85.799609 48.924219 85.037109 48.949219 z' />
+            </svg>
```
[the remark this answers is not in the document]
Comment (Julian-Caleb) [reply src/component/group-information-management/MembersTable.tsx:82]:
```suggestion
            <BsCheckCircle size={24} color="#4909B3" /> 
            ) : (
            <BsCheckCircle size={24} color="#9B9B9B" /> 
```

File: src/component/group-information-management/MembersTable.tsx [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+                    src='/images/see.png'
+                    alt='see'
+                    cursor='pointer'
+                    onClick={() =>
+                      toggleTooltip(
+                        { rowIndex, columnIndex: 6 },
+                        TooltipType.Attendance
+                      )
+                    }
+                  />
```
[the remark this answers is not in the document]
Comment (Julian-Caleb) [reply src/component/group-information-management/MembersTable.tsx:296]:
```suggestion
                  <FaEye
                    size={24}
                    color="#4b19a7"
                    cursor="pointer"
                    onClick={() => toggleTooltip({ rowIndex, columnIndex: 6 }, TooltipType.Attendance)}
                  />
```

File: src/component/group-information-management/MembersTable.tsx [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+                    src='/images/see.png'
+                    alt='see'
+                    cursor='pointer'
+                    onClick={() =>
+                      toggleTooltip(
+                        { rowIndex, columnIndex: 5 },
+                        TooltipType.Task
+                      )
+                    }
+                  />
```
[the remark this answers is not in the document]
Comment (Julian-Caleb) [reply src/component/group-information-management/MembersTable.tsx:251]:
```suggestion
                  <FaEye
                    size={24}
                    color="#4b19a7"
                    cursor="pointer"
                    onClick={() => toggleTooltip({ rowIndex, columnIndex: 5 }, TooltipType.Task)}
                  />
```

File: src/component/group-information-management/MembersTable.tsx [hunk 10]
```diff
@@ -0,0 +1,313 @@
...
+          ) : (
+            <svg
+              xmlns='http://www.w3.org/2000/svg'
+              viewBox='0 0 128 128'
+              fill='#9B9B9B'
+              width='24px'
+              height='24px'
+            >
+              <path d='M 64 6 C 32 6 6 32 6 64 C 6 96 32 122 64 122 C 96 122 122 96 122 64 C 122 32 96 6 64 6 z M 64 12 C 92.7 12 116 35.3 116 64 C 116 92.7 92.7 116 64 116 C 35.3 116 12 92.7 12 64 C 12 35.3 35.3 12 64 12 z M 85.037109 48.949219 C 84.274609 48.974219 83.500391 49.300391 82.900391 49.900391 L 62 71.599609 L 51.099609 59.900391 C 49.999609 58.700391 48.100391 58.599219 46.900391 59.699219 C 45.700391 60.799219 45.599219 62.700391 46.699219 63.900391 L 59.800781 78 C 60.400781 78.6 61.1 79 62 79 C 62.8 79 63.599219 78.699609 64.199219 78.099609 L 87.199219 54 C 88.299219 52.8 88.299609 50.900781 87.099609 49.800781 C 86.549609 49.200781 85.799609 48.924219 85.037109 48.949219 z' />
+            </svg>
```
[the remark this answers is not in the document]
Comment (Julian-Caleb) [reply src/component/group-information-management/MembersTable.tsx:114]:
```suggestion
            <BsCheckCircle size={24} color="#4909B3" /> 
          ) : (
            <BsCheckCircle size={24} color="#9B9B9B" /> 
```

**Comment by msyahrulsp** [comment]
Sip2, commit aja ya kalau udah pada diresolve. Kalau masih bingung, boleh tanya aja langsung ✌️

File: src/pages/group-information.tsx [hunk 11]
```diff
@@ -1,7 +1,148 @@
...
+          <>
+            {/* Mentor boxes*/}
+            <Box mt={4}>
+              <Text color='black' fontWeight='700' fontSize='20px'>
+                Mentor:
+              </Text>
+              <Flex justifyContent='space-between' flexWrap='wrap'>
+                {DUMMY_MENTORS.map((mentor, index) => (
+                  <MentorBox key={index} mentor={mentor} />
+                ))}
```
[the remark this answers is not in the document]
Comment (Julian-Caleb) [reply src/pages/group-information.tsx:132]: Cara manggilnya ke page bagaimana ya Kak?

[the remark this answers is not in the document]
Comment (msyahrulsp) [reply src/pages/group-information.tsx:132]: Hmm, ku belum cek lagi nama fungsi di benya apa. Paling kalua gimananya, bisa liat yg dari [dikpus](https://github.com/KATITB2023/dikpus-info/tree/main/src/component)

[the remark this answers is not in the document]
Comment (msyahrulsp) [reply src/pages/group-information.tsx:132]: image

kurleb gitu

**Comment by Julian-Caleb** [comment]
Maaf Kak, yang berhubungan dengan database sepertinya tidak bisa selesai malam ini karena saya masih bingung

**Pull request merged**